src/components/scene/background.js

```diff
@@ -1,49 +1,212 @@
-/* global THREE */
+/* global THREE, XRWebGLBinding */
 var register = require('../../core/component').registerComponent;
 var COMPONENTS = require('../../core/component').components;
 
 module.exports.Component = register('background', {
   schema: {
-    color: {type: 'color', default: 'black'},
-    transparent: {default: false},
-    generateEnvironment: {default: true}
+    color: { type: 'color', default: 'black' },
+    transparent: { default: false },
+    generateEnvironment: { default: true },
+    environmentUpdateFrequency: { default: 0 },
+    directionalLight: { type: 'selector' }
   },
   init: function () {
+    var self = this;
+
     this.cubeRenderTarget = new THREE.WebGLCubeRenderTarget(128, { format: THREE.RGBFormat, generateMipmaps: true, minFilter: THREE.LinearMipmapLinearFilter });
+    this.lightProbeTarget = new THREE.WebGLCubeRenderTarget(16, { format: THREE.RGBFormat, generateMipmaps: false });
     this.cubeCamera = new THREE.CubeCamera(1, 100000, this.cubeRenderTarget);
     this.needsEnvironmentUpdate = true;
+    this.timeSinceUpdate = 0;
+
+    this.el.sceneEl.addEventListener('enter-vr', function () {
+      var renderer = self.el.renderer;
+      var session = renderer.xr.getSession();
+      if (
+        session.requestLightProbe && self.el.sceneEl.is('ar-mode')
+      ) {
+        self.startLightProbe();
+      }
+    });
+
+    this.el.sceneEl.addEventListener('exit-vr', function () {
+      self.stopLightProbe();
+    });
+  },
+  stopLightProbe: function () {
+    var data = this.data;
+    var scene = this.el.sceneEl.object3D;
+    this.xrLightProbe = null;
+
+    this.probeLight.components.light.light.intensity = 0;
+    if (this.__ownDirectionalLight) {
+      this.__ownDirectionalLight.components.light.light.intensity = 0;
```

Comment:
can remove the double underscore `__` for consistency. Code base doesn't use variable naming for visibility